packages/@aws-cdk/aws-docdb/lib/cluster.ts

```diff
@@ -0,0 +1,454 @@
+      // Admin
+      masterUsername: secret ? secret.secretValueFromJson('username').toString() : props.masterUser.username,
+      masterUserPassword: secret
+        ? secret.secretValueFromJson('password').toString()
+        : props.masterUser.password!.toString(),
+      // Backup
+      backupRetentionPeriod: props.backup && props.backup.retention && props.backup.retention.toDays(),
+      preferredBackupWindow: props.backup && props.backup.preferredWindow,
+      preferredMaintenanceWindow: props.preferredMaintenanceWindow,
+      // Encryption
+      kmsKeyId: props.kmsKey && props.kmsKey.keyArn,
+      storageEncrypted: props.kmsKey ? true : props.storageEncrypted,
+    });
+
+    if (securityGroup instanceof ec2.SecurityGroup) {
+      cluster.addDependsOn(securityGroup.node.defaultChild as CfnResource);
+    }
+
+    cluster.applyRemovalPolicy(props.removalPolicy, {
+      applyToUpdateReplacePolicy: true
+    });
+
+    this.clusterIdentifier = cluster.ref;
+    this.clusterResourceIdentifier = cluster.attrClusterResourceId;
+
+    const port = Token.asNumber(cluster.attrPort);
+    this.clusterEndpoint = new Endpoint(cluster.attrEndpoint, port);
+    this.clusterReadEndpoint = new Endpoint(cluster.attrReadEndpoint, port);
+
+    if (secret) {
+      this.secret = secret.addTargetAttachment('AttachedSecret', {
+        target: this
+      });
+    }
+
+    // Create the instances
+    const instanceCount = props.instances != null ? props.instances : DatabaseCluster.DEFAULT_NUM_INSTANCES;
+    if (instanceCount < 1) {
+      throw new Error('At least one instance is required');
+    }
+
+    for (let i = 0; i < instanceCount; i++) {
+      const instanceIndex = i + 1;
+
+      const instanceIdentifier = props.instanceIdentifierBase != null ? `${props.instanceIdentifierBase}${instanceIndex}` :
+              props.dbClusterName != null ? `${props.dbClusterName}instance${instanceIndex}` :
+              undefined;
+
+      const instance = new CfnDBInstance(this, `Instance${instanceIndex}`, {
+        // Link to cluster
+        dbClusterIdentifier: cluster.ref,
+        dbInstanceIdentifier: instanceIdentifier,
+        // Instance properties
+        dbInstanceClass: databaseInstanceType(props.instanceProps.instanceType),
+      });
+
+      instance.applyRemovalPolicy(props.removalPolicy, {
+        applyToUpdateReplacePolicy: true
+      });
+
+      // We must have a dependency on the NAT gateway provider here to create
+      // things in the right order.
+      instance.node.addDependency(internetConnectivityEstablished);
+
+      this.instanceIdentifiers.push(instance.ref);
+      this.instanceEndpoints.push(new Endpoint(instance.attrEndpoint, port));
+    }
+
+    this.connections = new ec2.Connections({
+      defaultPort: ec2.Port.tcp(port),
+      securityGroups: [securityGroup]
+    });
+
+    this.node.defaultChild = cluster;
```

Comment:
Looks like this can be removed. I must have borrowed this from another package in the CDK repo. I traced this and found that it is not necessary because the cluster already uses the `"Resource"` construct ID.

See https://github.com/aws/aws-cdk/blob/cbe735e4c5a740c7e6f304bcd9aa867e4b44a9d8/packages/%40aws-cdk/core/lib/construct-compat.ts#L323-L343